python/tskit/metadata.py

```diff
@@ -568,7 +578,10 @@ def __init__(self, schema: Optional[Mapping[str, Any]]) -> None:
             self._validate_row = TSKITMetadataSchemaValidator(schema).validate
             self.encode_row = codec_instance.encode
             self.decode_row = codec_instance.decode
-            self.empty_value = {}
+            if "type" in schema and "null" in schema["type"]:
```

Comment:
Before I had this only putting the empty value as None if there were required attributes. I think this is wrong, and should be this way - None is "empty" if allowed.